### PR TITLE
adding rescuetime-focus-session-trigger

### PR DESCRIPTION
**Pull request by tuortheblessed**
## Description

Commands to start and stop a Focus Session on the RescueTime desktop or web apps. Simple API call for each command using a users personalized API Key.

## Screencast

image

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed in the `metadata` folder

### CODE REVIEW

File: extensions/rescuetime-focus-session-trigger/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+# RescueTime Focus Session Trigger Changelog
+
+## Added RescueTime Focus Session Trigger - (2022-3-19)
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/CHANGELOG.md:3]:
```suggestion
## [Added RescueTime Focus Session Trigger] - 2022-3-19
```
New format in the [changelog](https://developers.raycast.com/basics/prepare-an-extension-for-store#version-history)

File: extensions/rescuetime-focus-session-trigger/package.json [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      "type": "password",
+      "required": true,
+      "title": "RescueTime API key",
+      "placeholder": "https://www.rescuetime.com/anapi/manage"
+    }
+  ],
+  "commands": [
+    {
+      "name": "startFocusTime",
+      "title": "Start Focus Session",
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/package.json:25]:
```suggestion
      "title": "Start Focus Session",
      "subtitle": "RescueTime",
```

File: extensions/rescuetime-focus-session-trigger/package.json [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+          "type": "textfield",
+          "required": true,
+          "title": "Default Focus Session Duration",
+          "default": "25"
+        }
+      ]
+    },
+    {
+      "name": "endFocusTime",
+      "title": "End Focus Session",
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/package.json:41]:
```suggestion
      "title": "End Focus Session",
      "subtitle": "RescueTime",
```

File: extensions/rescuetime-focus-session-trigger/package-lock.json [hunk 4]
```diff
@@ -0,0 +1,8087 @@
+{
+  "name": "rescueTimeFocusSessionTrigger",
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/package-lock.json:2]:
```suggestion
  "name": "rescuetime-focus-session-trigger",
```
Just to match package.json

File: extensions/rescuetime-focus-session-trigger/package-lock.json [hunk 5]
```diff
@@ -0,0 +1,8087 @@
...
+  "lockfileVersion": 2,
+  "requires": true,
+  "packages": {
+    "": {
+      "name": "rescueTimeFocusSessionTrigger",
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/package-lock.json:7]:
```suggestion
      "name": "rescuetime-focus-session-trigger",
```

File: extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import { getPreferenceValues, showToast, Toast } from "@raycast/api";
+import got from "got";
+
+// TODO: implement changelog.md file?
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts:4]: Should be correct when changing format

File: extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+// re-do icon to meet guidelines
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts:5]: Is there anything wrong with it?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @tuortheblessed 👋 

Thanks for your contribution 🔥

I left some feedback for you to address 💪 

Do you have time to add [screenshots](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots) too?

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

File: extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts (hunk 6 above)
Comment (tuortheblessed) [reply extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts]: just missed removing these comments from developing.

**Comment by tuortheblessed** [comment]
@pernielsentikaer Appreciate the review and comments. Committed, ready for review.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @tuortheblessed 👋 

Sorry for the long response time, will have an extra eye on this 😄 


Thanks for the update, much appriciated!

A few questions:

- Do you have time to provide [screenshots](https://developers.raycast.com/basics/prepare-an-extension-for-store#screenshots)?

- I tried to generate a key, but it seems like it's read only all the time, is it because it's a premium feature somehow? 

7xi2OgoyBfg6ZOtO

Really looking forward to testing it out again 💪 

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by tuortheblessed** [comment]
@pernielsentikaer 

- I certainly could do screenshots, but the effort involved seemed to outweigh the benefit for something so simple. But let me know if they are desired.
- Yes, the Focus Session is a premium feature. I call that out in the beginning of the ReadMe, let me know if I need to make that more explicit.

File: extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+    APIkey: string;
+    duration: string;
+  }
+
+  const preferences = getPreferenceValues<Preferences>();
+
+  if (isNaN(parseInt(preferences.duration, 10))) {
+    showToast({
+      style: Toast.Style.Failure,
+      title: "Default Focus Session Durration Preferences needs to be a number",
```
Comment (pernielsentikaer) [inline extensions/rescuetime-focus-session-trigger/src/startFocusTime.ts:21]:
```suggestion
      title: "Failed to start Focus Session",
      message: "Default focus session duration needs to be a number",
```
Could we rephrase it like this? Corrected a spelling error and added the text as a message

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @tuortheblessed 👋 

No, it's just me that didn't catch it in the README; there is no need to change anything here 😄 

You're also right about the screenshots, and it doesn't make sense since this is a no-view command 💪 

Only found one minor correction before merging. Could you look into this?

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Review by pernielsentikaer — APPROVED** [review]
🥳

**Pull request merged**